src/index.js

```diff
@@ -146,9 +152,9 @@ export default class Pipeliner {
 
   /**
    * Runs the specified pipeline, passing the arguments to each task.
-   * @param  {string}    pipeline the pipeline to run
+   * @param  {string}    pipeline The name of the pipeline to run
    * @param  {...object} args     Arguments to pass to the pipeline tasks
-   * @return {Promise}             A promise that is executed when the pipeline completes.
+   * @return {Promise}            A promise that is fulfilled when the pipeline completes; it is rejected if any task in the pipeline fails (throws an error or returns a promise that is rejected)
```

Comment:
Almost.  Promise.resolve() can return two things: 1) if the wrapped function returns a promise, Promise.resolve() returns the same promise (passes it through); 2) if the wrapped function returns anything else, Promise.resolve() wraps that return value in a promise that resolves to the value. The gotcha is that Promise.resolve() will always return a promise that resolves, even if the returning function throws an exception.  This means that Promise.resolve(...).catch() will never fire, no matter what happens inside the wrapped function.  However, Promise.try() will act just like the native try(), returning a Promise that resolves to the returned value (which can be another promise) OR a promise that rejects if the wrapped function throws an exception.

In either case, Promise.mapSeries will process the passed array of functions in series, handling both synchronous and asynchronous errors the same.  The problem is Promise.resolve() will never result in errors (resolve, by definition is the non-error state).

I think in the original code, this should be classified as a bug, since we would probably want to at least catch an exception in the pipeline stage.  However, I think all the tasks I've been writing return promises anyhow, so it's not come up (though Promise.resolve has bitten us elsewhere when we expected to get an exception but instead got silently failing code...)